src/main/io/cms.c

```diff
@@ -0,0 +1,1213 @@
+#ifdef OSD
+            if (p->data) {
+                uint32_t address = (uint32_t)p->data;
+                uint16_t *val;
+
+                val = (uint16_t *)address;
+
+                if (key == KEY_RIGHT)
+                    *val |= VISIBLE_FLAG;
+                else
+                    *val %= ~VISIBLE_FLAG;
+                p->changed = true;
+            }
+#endif
+            break;
+        case OME_UINT8:
+        case OME_FLOAT:
+            if (p->data) {
+                OSD_UINT8_t *ptr = p->data;
+                if (key == KEY_RIGHT) {
+                    if (*ptr->val < ptr->max)
+                        *ptr->val += ptr->step;
+                }
+                else {
+                    if (*ptr->val > ptr->min)
+                        *ptr->val -= ptr->step;
+                }
+                p->changed = true;
+            }
+            break;
+        case OME_TAB:
+            if (p->type == OME_TAB) {
+                OSD_TAB_t *ptr = p->data;
+
+                if (key == KEY_RIGHT) {
+                    if (*ptr->val < ptr->max)
+                        *ptr->val += 1;
+                }
+                else {
+                    if (*ptr->val > 0)
+                        *ptr->val -= 1;
+                }
+                if (p->func)
+                    p->func(p->data);
+                p->changed = true;
+            }
+            break;
+        case OME_INT8:
+            if (p->data) {
+                OSD_INT8_t *ptr = p->data;
+                if (key == KEY_RIGHT) {
+                    if (*ptr->val < ptr->max)
+                        *ptr->val += ptr->step;
+                }
+                else {
+                    if (*ptr->val > ptr->min)
+                        *ptr->val -= ptr->step;
+                }
+                p->changed = true;
+            }
+            break;
+        case OME_UINT16:
+            if (p->data) {
+                OSD_UINT16_t *ptr = p->data;
+                if (key == KEY_RIGHT) {
+                    if (*ptr->val < ptr->max)
+                        *ptr->val += ptr->step;
+                }
+                else {
+                    if (*ptr->val > ptr->min)
+                        *ptr->val -= ptr->step;
+                }
+                p->changed = true;
+            }
+            break;
+        case OME_INT16:
+            if (p->data) {
+                OSD_INT16_t *ptr = p->data;
+                if (key == KEY_RIGHT) {
+                    if (*ptr->val < ptr->max)
+                        *ptr->val += ptr->step;
+                }
+                else {
+                    if (*ptr->val > ptr->min)
+                        *ptr->val -= ptr->step;
+                }
+                p->changed = true;
+            }
+            break;
+        case OME_Poll_INT16:
+        case OME_Label:
+        case OME_END:
+            break;
+    }
+    return res;
+}
+
+static void simple_ftoa(int32_t value, char *floatString)
```

Comment:
It was inherited from the original code without questioning :) 

As I examined it, I found out that it's a fixed point conversion; the value to be displayed is uint8_t.